src/behavior-not-considered-unsafe.md

```diff
@@ -36,4 +36,21 @@ semantics.
 See [RFC 560] for error conditions, rationale, and more details about
 integer overflow.
 
+##### Logic errors
+
+Safe code may impose extra logical constraints that can be checked
+at neither compile-time nor runtime. If a program breaks such
+a constraint, the behavior may be unspecified but will not result in
+undefined behavior. This could include panics, incorrect results,
+aborts, and non-termination. The behavior may also differ between
+runs, builds or kinds of build.
+
+For example, implementing both `Hash` and `Eq` requires that values
+considered equal have equal hashes. Another example are data structures
+like `BinaryHeap`, `BTreeMap`, `BTreeSet`, `HashMap` and `HashSet`
+which describe constraints on the modification of their keys while
+they are in the data structure. Violating such constraints is not
+considered unsafe, yet the program is considered erroneous and
+its behavior, unpredictable.
```

Comment:
I was mainly putting it there in place of the verb (since I omitted it to avoid repeating it). I have now tried to look up if that is done or not in English (like in Spanish), but it looks like it is only done when the clauses are split by a semicolon and no conjunction? I am curious now... :-)